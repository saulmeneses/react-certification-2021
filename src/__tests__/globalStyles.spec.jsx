import React from 'react';
import { ThemeProvider } from 'styled-components';
import renderer from 'react-test-renderer';

import GlobalStyle from '../globalStyles';

describe('GlobalStyle', () => {
  it('renders and match theme styles', () => {
    const theme = ['body', 'appBar', 'card', 'cardText'].reduce(
      (acc, key) => ({
        ...acc,
        [key]: {
          color: 'white',
          backgroundColor: 'gray',
        },
      }),
      {}
    );

    const tree = renderer
      .create(
        <ThemeProvider theme={theme}>
          <GlobalStyle />
        </ThemeProvider>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
