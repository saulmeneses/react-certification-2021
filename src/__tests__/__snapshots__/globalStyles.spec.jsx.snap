// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GlobalStyle renders and match theme styles 1`] = `null`;
